google/cloud/storage/tests/async_client_integration_test.cc

```diff
@@ -405,6 +405,99 @@ TEST_F(AsyncClientIntegrationTest, StartBufferedUploadMultiple) {
   EXPECT_EQ(metadata->size(), kBlockCount * kBlockSize);
 }
 
+TEST_F(AsyncClientIntegrationTest, RewriteObject) {
+  auto o1 = MakeRandomObjectName();
+  auto o2 = MakeRandomObjectName();
+
+  auto async = AsyncClient();
+  auto constexpr kBlockSize = 4 * 1024 * 1024;
+  auto insert = async
+                    .InsertObject(bucket_name(), o1, MakeRandomData(kBlockSize),
+                                  gcs::IfGenerationMatch(0))
+                    .get();
+  ASSERT_STATUS_OK(insert);
+  ScheduleForDelete(*insert);
+
+  // Start a rewrite, but limit each iteration to a small number of bytes, to
+  // force multiple calls.
+  storage::ObjectMetadata metadata;
+  AsyncRewriter rewriter;
+  AsyncToken token;
+  std::tie(rewriter, token) =
+      async.StartRewrite(bucket_name(), o1, bucket_name(), o2,
+                         storage::MaxBytesRewrittenPerCall(1024 * 1024));
+  while (token.valid()) {
+    auto rt = rewriter.Iterate(std::move(token)).get();
+    ASSERT_STATUS_OK(rt);
+    RewriteObjectResponse response;
+    AsyncToken t;
+    std::tie(response, t) = *std::move(rt);
+    token = std::move(t);
+    if (!response.metadata.has_value()) continue;
+    ScheduleForDelete(*response.metadata);
+    metadata = *response.metadata;
+    break;
```

Comment:
nit: if `token.valid() == false`, we don't need this break.

Reply:
Good point, replaced with a `EXPECT_FALSE()` because it is something we should test for.